### PR TITLE
Added github actions workflow to check linting

### LINKED ISSUE #72: Add github actions workflow to check linting
Add a github actions workflow that checks if the committed files are passing the linting checking put in place.

Requirements:
- [ ] Add a `.github/workflows/linting.yml` file that runs linting checks for any push/PR.

It should run the lint script declared [here](https://github.com/adityabisoi/REST-fetch/blob/main/package.json#L9)

### PR DESCRIPTION
**Pull request by AshuKV**
# Related Issue

-   closes #72 

# Proposed Changes

-   change 1
-   change 2

# Additional Info

-   any additional information or context

# Checklist

-   [x] Tests
-   [x] Translations
-   [x] Documentation

# Screenshots

|         Original          |         Updated          |
| :-----------------------: | :----------------------: |
| ** original screenshot ** | ** updated screenshot ** |

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+env:
```
Comment (ShauryaAg) [inline .eslintrc.yml:1]: We don't need this. We already have a `.eslintrc` config file

File: package.json [hunk 2]
```diff
@@ -33,12 +33,14 @@
         "@types/mongoose": "^5.10.4",
         "@types/morgan": "^1.9.2",
         "@types/node": "^14.14.37",
-        "@typescript-eslint/eslint-plugin": "^4.20.0",
-        "eslint": "^7.23.0",
+        "@typescript-eslint/eslint-plugin": "^4.21.0",
+        "@typescript-eslint/parser": "^4.21.0",
+        "eslint": "^7.24.0",
```
Comment (ShauryaAg) [inline package.json:38]: Why were these changes made?

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+on: [push]
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: stefanoeb/eslint-action@1.0.2
+        with:
+          files: ./src/**/*.ts
```
Comment (ShauryaAg) [inline .github/workflows/linting.yml:12]: Instead of using this action. We can just simply use `npm run lint`

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by ShauryaAg** [comment]
You may use [this](https://docs.github.com/en/actions/guides/building-and-testing-nodejs) and [this](https://sviat-kuzhelev.medium.com/progressive-front-end-ci-on-github-actions-817a29938e4f) for reference.

**Review by AshuKV — COMMENTED** [review]
I have updated the PR. It can be merged now.

File: package.json [hunk 4]
```diff
@@ -35,11 +35,13 @@
         "@types/morgan": "^1.9.2",
         "@types/node": "^14.14.37",
         "@typescript-eslint/eslint-plugin": "^4.20.0",
+        "@typescript-eslint/parser": "^4.21.0",
         "eslint": "^7.23.0",
         "eslint-config-prettier": "^8.1.0",
         "eslint-plugin-import": "^2.22.1",
         "eslint-plugin-json": "^2.1.2",
         "eslint-plugin-prettier": "^3.3.1",
+        "eslint-plugin-react": "^7.23.2",
```
Comment (ShauryaAg) [inline package.json:44]: Why is this change needed?

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Review by ShauryaAg — APPROVED** [review]

**Pull request merged**